onnxruntime/test/shared_lib/custom_op_utils.cc

```diff
@@ -168,6 +168,94 @@ void MyCustomKernelWithOptionalInput::Compute(OrtKernelContext* context) {
   }
 }
 
+void MyCustomStringLengthsKernel::Compute(OrtKernelContext* context) {
+  Ort::KernelContext kcontext(context);
+  std::array<const int64_t, 1> output_shape = {1};
+
+  const size_t num_inputs = kcontext.GetInputCount();
+
+  // Each output is set to the length of the corresponding input string.
+  for (size_t i = 0; i < num_inputs; ++i) {
+    auto input = kcontext.GetInput(i);
+    auto output = kcontext.GetOutput(i, output_shape.data(), output_shape.size());
+    int64_t* str_len_ptr = output.GetTensorMutableData<int64_t>();
+
+    *str_len_ptr = input.GetStringTensorElementLength(0);
+  }
+}
+
+void AddInputForCustomStringLengthsKernel(std::string input_str, OrtAllocator* allocator,
+                                          std::vector<Ort::Value>& ort_inputs, std::vector<std::string>& input_names,
+                                          std::vector<std::string>& output_names,
+                                          std::vector<std::vector<int64_t>>& expected_dims,
+                                          std::vector<std::vector<int64_t>>& expected_outputs) {
+  const size_t index = ort_inputs.size();
+
+  std::array<int64_t, 1> input_dims = {1};
+  Ort::Value& ort_value = ort_inputs.emplace_back(
+      Ort::Value::CreateTensor(allocator, input_dims.data(), input_dims.size(),
+                               ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING));
+  std::ostringstream oss(std::ostringstream::ate);
+
+  oss.str("input_");
+  oss << index;
+  input_names.emplace_back(oss.str());
+
+  oss.str("output_");
+  oss << index;
+  output_names.emplace_back(oss.str());
+
+  expected_dims.push_back({1});
+  expected_outputs.push_back({static_cast<int64_t>(input_str.size())});
+  ort_value.FillStringTensorElement(input_str.data(), 0);
+}
+
+void MyCustomEchoReversedArgsKernel::Compute(OrtKernelContext* context) {
+  Ort::KernelContext kcontext(context);
+  std::array<const int64_t, 1> output_shape = {1};
```

Comment:
>const [](http://example.com/codeflow?start=13&length=5)

Since it is an example, a better way of doing it would be to declare output_shape constexpr and element type leave as non-const, consistent with other STD containers constness.

Reply:
Updated to use `constexpr`